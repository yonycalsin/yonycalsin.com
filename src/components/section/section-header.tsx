import * as React from 'react'
import { Box, Heading } from '@chakra-ui/react'

export interface SectionHeaderProps {
  title: string
  actionHref?: string
  actionLabel?: string
  actionHrefExternal?: boolean
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  actionComponent?: any
  children?: React.ReactNode
}

export function SectionHeader(props: SectionHeaderProps) {
  const {
    title,
    actionHref,
    actionLabel,
    actionComponent: ActionComponent = 'a',
    actionHrefExternal = false,
    children,
  } = props

  return (
    <div>
      <Box display="flex" alignItems="center" justifyContent="space-between" position="relative">
        <Heading size="lg" fontWeight="bold">
          {title}
        </Heading>
        {actionHref ? (
          <ActionComponent href={actionHref} target={actionHrefExternal ? '_blank' : null}>
            {actionLabel}
          </ActionComponent>
        ) : null}
      </Box>
      {children}
    </div>
  )
}
