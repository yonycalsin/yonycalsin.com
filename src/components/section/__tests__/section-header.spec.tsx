import { render, screen } from '@testing-library/react'

import { SectionHeader, SectionHeaderProps } from '../section-header'

const defaultSectionHeaderProps: SectionHeaderProps = {
  title: 'Customers',
}

const setup = (props?: Partial<SectionHeaderProps>) => {
  return render(<SectionHeader {...props} {...defaultSectionHeaderProps} />)
}

describe('Components / SectionHeader', () => {
  it('renders the section header', () => {
    const view = setup()

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders the section header with anchor', () => {
    const view = setup({
      actionHref: '/customers',
      actionLabel: 'See all',
    })

    const actionLink = screen.queryByRole('link', { name: /see all/i })

    expect(actionLink).toBeInTheDocument()

    expect(actionLink).not.toHaveAttribute('target', '_blank')

    expect(view.baseElement).toMatchSnapshot()
  })

  it('renders the section header with external anchor', () => {
    const view = setup({
      actionHref: '/customers',
      actionLabel: 'See all',
      actionHrefExternal: true,
    })

    const actionLink = screen.queryByRole('link', { name: /see all/i })

    expect(actionLink).toBeInTheDocument()

    expect(actionLink).toHaveAttribute('target', '_blank')

    expect(view.baseElement).toMatchSnapshot()
  })
})
