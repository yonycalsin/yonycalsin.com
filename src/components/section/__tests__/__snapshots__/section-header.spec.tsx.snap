// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionHeader renders the section header 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex items-center justify-between relative"
      >
        <h1
          class="font-bold font-heading text-2xl"
        >
          Customers
        </h1>
      </div>
    </div>
  </div>
</body>
`;

exports[`SectionHeader renders the section header with anchor 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex items-center justify-between relative"
      >
        <h1
          class="font-bold font-heading text-2xl"
        >
          Customers
        </h1>
        <a
          href="/customers"
        >
          See all
        </a>
      </div>
    </div>
  </div>
</body>
`;

exports[`SectionHeader renders the section header with external anchor 1`] = `
<body>
  <div>
    <div>
      <div
        class="flex items-center justify-between relative"
      >
        <h1
          class="font-bold font-heading text-2xl"
        >
          Customers
        </h1>
        <a
          href="/customers"
          target="_blank"
        >
          See all
        </a>
      </div>
    </div>
  </div>
</body>
`;
